charts/bungee/templates/deployment.yaml

```diff
@@ -0,0 +1,170 @@
+        {{- if .Values.bungeeCord.jenkinsJobId }}
+        - name: BUNGEE_JOB_ID
+          value: {{ .Values.bungeeCord.jenkinsJobId }}
+        {{- end }}
+        {{- if .Values.bungeeCord.jenkinsBaseUrl }}
+        - name: BUNGEE_BASE_URL
+          value: {{ .Values.bungeeCord.jenkinsBaseUrl }}
+        {{- end }}
+        {{- if .Values.bungeeCord.jarRevision }}
+        - name: BUNGEE_JAR_REVISION
+          value: {{ .Values.bungeeCord.jarRevision }}
+        {{- end }}
+        {{- end }}
+        {{- if .Values.plugins }}
+        - name: PLUGINS
+          value: {{ include "helm-toolkit.utils.joinListWithComma" .Values.plugins }}
+        {{- end }}
+        - name: MEMORY
+          value: {{ .Values.bungeeCord.memory | quote }}
+        - name: JVM_OPTS
+          value: {{ .Values.bungeeCord.jvmOpts | quote }}
+
+        {{- if .Values.bungeeCord.rcon.enabled }}
+        - name: ENABLE_RCON
+          value: "true"
+        - name: RCON_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: {{ template "bungee.fullname" . }}
+              key: rcon-password
+        {{- else }}
+        - name: ENABLE_RCON
+          value: "false"
+        {{- end }}
+
+        {{- range $key, $value := .Values.extraEnv }}
+        - name: {{ $key }}
+          value: {{ $value | quote }}
+        {{- end }}
+
+        ports:
+        - name: bungee
+          containerPort: 25577
+          protocol: TCP
+        {{- if .Values.bungeeCord.rcon.enabled }}
+        - name: rcon
+          containerPort: 25575
+          protocol: TCP
+        {{- end }}
+        volumeMounts:
+        - name: datadir
+          mountPath: /server
+        {{- if .Values.bungeeCord.config }}
+        - name: config
+          mountPath: {{ .Values.bungeeCord.configFilePath }}
+          subPath: config.yml
+        {{- end }}
+        {{- range .Values.extraVolumes }}
+        {{- toYaml .volumeMounts | nindent 8 }}
+        {{- end }}
+      volumes:
+      - name: datadir
+      {{- if .Values.persistence.dataDir.enabled }}
+        persistentVolumeClaim:
+        {{- if .Values.persistence.dataDir.existingClaim }}
+          claimName: {{ .Values.persistence.dataDir.existingClaim }}
+        {{- else }}
+          claimName: {{ template "bungee.fullname" . }}-datadir
+        {{- end }}
+      {{- else }}
+        emptyDir: {}
+      {{- end }}
+      {{- if .Values.bungeeCord.config }}
+      - name: config
+        configMap:
+          name: {{ include "bungee.fullname" . }}-config
+      {{- end }}
+      {{- range .Values.extraVolumes }}
+      {{- toYaml .volumes | nindent 6 }}
+      {{- end }}
+    {{- if .Values.nodeSelector }}
+      nodeSelector: {{ toYaml .Values.nodeSelector | nindent 8 }}
+    {{- end }}
+    {{- if .Values.affinity }}
+      affinity: {{ toYaml .Values.affinity | nindent 8 }}
+    {{- end }}
+    {{- if .Values.tolerations }}
+      tolerations: {{ toYaml .Values.tolerations | nindent 8 }}
+    {{- end }}
```

Comment:
> Wondering if it's worth adding nodeSelector, affinity, and tolerations to the values.yaml and remove the if/end logic from the end of the deployment.

I see what you mean. Something like a pod customizations section?

Reply:
I'm happy with this for now, it can be iterated on later, it's out of scope of this PR I think